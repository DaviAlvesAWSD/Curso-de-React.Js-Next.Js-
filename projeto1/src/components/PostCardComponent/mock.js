export const postCardPropsMock = {
  title: 'title 1',
  body: 'body 1',
  id: 1,
  cover: 'img/img.png'
};
