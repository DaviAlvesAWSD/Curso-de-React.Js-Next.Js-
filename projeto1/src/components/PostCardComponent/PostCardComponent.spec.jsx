import { PostCardComponent } from '.';
import { render, screen } from '@testing-library/react';
import { postCardPropsMock } from './mock';

const props = postCardPropsMock;

describe('<PostCardComponent />', () => {
  it('should render PostCardComponent correctly', () => {
    render(<PostCardComponent {...props} />);

    expect(screen.getByRole('img', { name: 'title 1' })).toHaveAttribute('src', 'img/img.png');

    expect(screen.getByRole('heading', { name: 'title 1' })).toBeInTheDocument();
    expect(screen.getByText('body 1')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCardComponent {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
