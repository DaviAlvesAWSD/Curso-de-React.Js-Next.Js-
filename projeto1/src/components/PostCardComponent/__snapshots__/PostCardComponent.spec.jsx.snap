// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCardComponent /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="title 1"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      title 1
       
    </h2>
    <p>
      body 1
    </p>
  </div>
</div>
`;
